### ISSUE TITLE
[turborepo] 

### ISSUE THREAD

**Issue by drorzp** [vercel/turbo#5496]
### What version of Turborepo are you using?

1.10.6

### What package manager are you using / does the bug impact?

npm

### What operating system are you using?

Windows

### Describe the Bug

when running npm run format ("format": "turbo run format") 
or any other item "lint": "turbo run lint" etc i see the below:

ERROR Unable to find Go binary. Please report this issue at https://github.com/vercel/turbo/issues and include your package manager and version along with the following information:   
        os=windows
        arch=x86_64
        turbo-version=1.10.6
        turbo-bin=C:\Users\xbbn46w\dev\admin-tool\node_modules\turbo\bin\downloaded-turbo-windows-64-turbo.exe
        go-turbo-bin=C:\Users\xbbn46w\dev\admin-tool\node_modules\turbo\bin\go-turbo.exe    

### Expected Behavior

all npm commands should pass with no error

### To Reproduce

can not 

### Reproduction Repo

can not

**Comment by gsoltis** [comment]
It appears that this is an error in the installation process. Can you provide details of how you install `turbo`?

Can you try installing optional dependencies along with `turbo`?

**Comment by codyfyi** [comment]
@gsoltis

We are also seeing this issue on our GitLab CI linux build machines: https://github.com/vercel/turbo/issues/5512#issuecomment-1679720117

```
npm run build
> turbo build
Error: Failed to execute turbo (Unable to locate Go binary).
ERROR Unable to find Go binary. Please report this issue at https://github.com/vercel/turbo/issues and include your package manager and version along with the following information:
        os=linux
        arch=x86_64
        turbo-version=1.10.12
        turbo-bin=/builds/.../{app}/node_modules/turbo/bin/downloaded-turbo-linux-64-turbo
        go-turbo-bin=/builds/.../{app}/node_modules/turbo/bin/go-turbo
```

Turbo is listed in package.json `dependencies` and installed using `npm install`. Then we run `npm run build` which throws the above error.

**Comment by dkobierski** [comment]
@codyfyi I had the same issue - downgrading to 1.10.10 works fine. They might have fixed the issue in 1.10.10 and re-introduce the it in 1.10.12.

**Comment by dios-david** [comment]
I see the same issue when running inside `node18-alpine` docker image:

```
ERROR Unable to find Go binary. Please report this issue at https://github.com/vercel/turbo/issues and include your package manager and version along with the following information:
        os=linux
        arch=x86_64
        turbo-version=1.10.12
        turbo-bin=/workspace/node_modules/turbo/bin/downloaded-turbo-linux-64-turbo
        go-turbo-bin=/workspace/node_modules/turbo/bin/go-turbo
```

**Comment by nathanhammond** [comment]
This should be addressed in the next release to `latest`, or you can migrate to `turbo@canary` now.

Addressed by #5695.
